test/Land/Model/freeze_thaw_alone.jl

```diff
@@ -134,43 +134,31 @@ using ClimateMachine.BalanceLaws:
         driver_config,
         ode_dt = dt,
     )
-    mygrid = solver_config.dg.grid
-    Q = solver_config.Q
-
-    ϑ_l_ind = varsindex(vars_state(m, Prognostic(), FT), :soil, :water, :ϑ_l)
-    θ_i_ind = varsindex(vars_state(m, Prognostic(), FT), :soil, :water, :θ_i)
-
-    all_data = Dict([k => Dict() for k in 1:n_outputs]...)
-    step = [1]
-    callback = GenericCallbacks.EveryXSimulationTime(
-        every_x_simulation_time,
-    ) do (init = false)
-        t = ODESolvers.gettime(solver_config.solver)
-        ϑ_l = Q[:, ϑ_l_ind, :]
-        θ_i = Q[:, θ_i_ind, :]
-        all_vars = Dict{String, Array}("t" => [t], "ϑ_l" => ϑ_l, "θ_i" => θ_i)
-        all_data[step[1]] = all_vars
-        step[1] += 1
+    state_types = (Prognostic(),)
+    all_data =
+        Dict[dict_of_nodal_states(solver_config, state_types; interp = true)]
+    time_data = FT[0]
+    callback = GenericCallbacks.EveryXSimulationTime(every_x_simulation_time) do
+        dons = dict_of_nodal_states(solver_config, state_types; interp = false)
```

Comment:
```suggestion
        dons = dict_of_nodal_states(solver_config, state_types; interp = true)
```
You'll want all of these to match

Reply:
good catch! Question on this - if we have an analytic solution, do we expect the output to match the analytic solution if we use interpolation?

Reply:
Hmm, I'm not sure if I understand. The interpolation is really just interpolating between the duplicate lgl points at the cell faces, so the solution quality should be maintained.

Reply:
ah ok, it works both ways. that makes sense to me!